src/react-modules/JSONUpload.tsx

```diff
@@ -1,37 +1,45 @@
 import { useState, useRef } from "react";
 import { ErrorModal } from "./ErrorModal";
-// import { BookmarkSquareIcon } from "@heroicons/react/20/solid";
+import { ShowcaseJSON } from "../types";
 
-function JSONUploadButton({ setShowcaseJSON }) {
-  // Error handling
+export const JSONUploadButton = ({
+  setShowcaseJSON,
+}: {
+  setShowcaseJSON: React.Dispatch<React.SetStateAction<ShowcaseJSON>>;
+}) => {
   const [showModal, setShowModal] = useState(false);
+  const fileInputRef = useRef<HTMLInputElement>(null);
 
-  const fileInputRef = useRef(null);
-
-  const handleFileUpload = (event) => {
-    const file = event.target.files[0];
+  const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
     if (file && file.type === "application/json") {
       const reader = new FileReader();
       reader.readAsText(file);
       reader.onload = () => {
         const fileContent = reader.result;
         console.log(fileContent);
+        console.log(typeof fileContent);
         try {
-          setShowcaseJSON(JSON.parse(`{"personas": ${fileContent}}`));
+          if (typeof fileContent === 'string') {
```

Comment:
this feature is not working at all on the original project

Reply:
Could you create a FIXME in the code base with a ticket?